File: src/test/java/nz/co/searchwellington/commentfeeds/detectors/DateRegexCommentFeedDetectorTest.java
package nz.co.searchwellington.commentfeeds.detectors;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

import java.net.MalformedURLException;
import java.net.URL;

public class DateRegexCommentFeedDetectorTest {
	
	private CommentFeedDetector commentFeedDetector = new DateRegexCommentFeedDetector();

	@Test
	public void shouldMatchUrlWithDateInThemAsTheseAreAlmostAlwaysCommentFeedsForSpecficPost() throws Exception {
		final URL url = new URL("http://www.blah.nz/something/2011/01/20/comments");
		assertTrue(commentFeedDetector.isValid(url));
	}

	@Test
	public void shouldNotGiveObviousFalsePositives() throws Exception {
		final URL url = new URL("http://www.blah.nz/something/comments");
		assertFalse(commentFeedDetector.isValid(url));
	}
	
}

File: src/test/scala/nz/co/searchwellington/commentfeeds/detectors/DateRegexCommentFeedDetectorTest.scala
package nz.co.searchwellington.commentfeeds.detectors

import nz.co.searchwellington.model.Newsitem
import org.junit.jupiter.api.Assertions.{assertFalse, assertTrue}
import org.junit.jupiter.api.Test

import java.net.URL

class DateRegexCommentFeedDetectorTest {

  private val commentFeedDetector = new DateRegexCommentFeedDetector

  @Test
  def shouldMatchUrlWithDateInThemAsTheseAreAlmostAlwaysCommentFeedsForSpecficPost(): Unit = {
    val source = Newsitem()
    val url = new URL("http://www.blah.nz/something/2011/01/20/comments")
    assertTrue(commentFeedDetector.isValid(url, source))
  }

  @Test
  def shouldNotGiveObviousFalsePositives(): Unit = {
    val source = Newsitem()
    val url = new URL("http://www.blah.nz/something/comments")
    assertFalse(commentFeedDetector.isValid(url, source))
  }

}